client/components/backup-retention-management/retention-options/retention-options-control.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import React from 'react';
+import RetentionOptionCard from './retention-option-card';
+import type { RetentionRadioOptionType } from '../types';
+
+interface RetentionOptionsControlProps {
+	retentionOptions: Record< number, RetentionRadioOptionType >;
+	retentionSelected?: number;
+	onChange: ( value: number ) => void;
+}
+
+const RetentionOptionsControl: React.FC< RetentionOptionsControlProps > = ( {
+	retentionOptions,
+	retentionSelected,
+	onChange,
+} ) => {
+	const [ selectedOption, setSelectedOption ] = useState( retentionSelected );
+
+	useEffect( () => {
+		if ( typeof selectedOption === 'number' ) {
+			onChange( selectedOption );
+		}
+	}, [ selectedOption, onChange ] );
+
+	const handleRetentionOptionChange = useCallback( ( value: number ) => {
+		setSelectedOption( value );
+	}, [] );
+
+	return (
+		<div className="retention-form__options">
+			{ Object.values( retentionOptions ).map( ( option ) => (
```

Comment:
If we're going to use `retentionOptions` as an array of values, should we reform the component to expect an array and use that instead of calling `Object.values()`  on an object every time?

Reply:
Agree, it makes sense. We just prepared a refactor for this.

Addressed in #73303